OpenRA.Mods.RA2/Traits/SpawnNeighboringActors.cs

```diff
@@ -0,0 +1,97 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	[Desc("This actor places other actors around itself, which keep connected.")]
+	public class SpawnNeighboringActorsInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[ActorReference]
+		[Desc("Types of actors to place. If multiple are defined, a random one will be selected.")]
+		public readonly HashSet<string> ActorTypes = new HashSet<string>();
+
+		[FieldLoader.Require]
+		[Desc("Locations to spawn the actors relative to the origin (top-left for buildings) of this actor.")]
+		public readonly CVec[] Locations = { };
+
+		public object Create(ActorInitializer init) { return new SpawnNeighboringActors(this, init.Self); }
+	}
+
+	public class SpawnNeighboringActors : INotifyKilled, INotifyOwnerChanged, INotifyActorDisposing, INotifyAddedToWorld, INotifySold
+	{
+		readonly SpawnNeighboringActorsInfo info;
+		readonly List<Actor> actors = new List<Actor>();
```

Comment:
I don't see items being added to this list (which could be a hashset? not sure if that'll really matter).
Given that, I don't understand how this trait actually keeps ownership/existence in sync.

Reply:
That is a bug. Fixed below.